File: problems/Is Subsequence/isSubsequence.cs
public class Solution {
    public bool IsSubsequence(string s, string t) {
        int patternIdx = 0;
        int textIdx = 0;

        while (s.Length > patternIdx && t.Length > textIdx) {
            if (s[patternIdx] == t[textIdx]) {
                ++patternIdx;
            }
            ++textIdx;
        }

        return s.Length == patternIdx;
    }
}

File: problems/Is Subsequence/isSubsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int sIdx = 0;
        int tIdx = 0;

        while (s.length() > sIdx && t.length() > tIdx) {
            if (s.charAt(sIdx) == t.charAt(tIdx)) {
                ++sIdx;
            }
            ++tIdx;
        }

        return sIdx == s.length();
    }
}
